### PR TITLE
Kevin

### PR DESCRIPTION
**Pull request by KRT0335**
BRUH Request
Adding tests from the "Kevin" branch.

### CODE REVIEW

**Review by AdamDKing — CHANGES_REQUESTED** [review]
Read the README, follow the style guide.

**Comment by KRT0335** [comment]
I updated the style

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java [hunk 1]
```diff
@@ -164,6 +235,18 @@ public void testGetAllResponses() {
     assertEquals(ResponseEntity.ok(listResponses), this.responseControllerImpl.getAllResponses());
   }
 
+  @Test
+  public void testDeleteResponse() {
+
+    doNothing().when(responseServiceMock).delete(10);
+    responseControllerImpl.deleteResponse(10);
+  }
```
Comment (AdamDKing) [inline src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: This doesn't seem to test anything -- when would this fail?

File: src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java [hunk 2]
```diff
@@ -495,18 +996,50 @@ public void getImageByIdTest() {
...
+  // byte[] byteArr = null;
+  // MockMultipartFile firstFile = new MockMultipartFile("data", "filename.txt", "text/plain",
+  // byteArr);
+  // MultipartFile mpf = firstFile;
+  // when(mockRequest.getFile("image")).thenReturn(firstFile);
+  //// when(mpf.getBytes()).thenReturn(null);
+  //
+  // imageServiceImpl.addImage(testQuestion1, mockRequest);
+  // }
+
```
Comment (AdamDKing) [inline src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java]: Remove commented out code -- if you want it back later there's always Git!

File: src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java [hunk 3]
```diff
@@ -972,36 +1498,47 @@ public void questionsQuestionNotFoundWhenUpdatingTagsExceptionTest() {
...
   @Test(expected = QuestionConflictException.class)
   public void questionsConflictExceptionWhenHighlightingResponseTest() {
-    when(questionRepositoryMock.findById(1)).thenReturn(Optional.of(testQuestion1));
-    when(questionRepositoryMock.save(testQuestion1)).thenThrow(DataIntegrityViolationException.class);
-    
+    // when(questionRepositoryMock.findById(1)).thenReturn(Optional.of(testQuestion1));
+    // when(questionServiceImpl.getById(1)).thenReturn((Question)
+    // when(questionRepositoryMock.findById(1)).thenReturn(Optional.of(testQuestion1)));
+    // when(questionServiceImpl.update(testQuestion1)).thenThrow(DataIntegrityViolationException.class);
+    // when(questionRepositoryMock.save(testQuestion1)).thenThrow(DataIntegrityViolationException.class);
```
Comment (AdamDKing) [inline src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java]: remove

**Review by AdamDKing — CHANGES_REQUESTED** [review]
Gitignore all the html reports you're generating... also the reports all have stack traces indicating some kind of failure, so fix whatever problem that is.  You're also pushing an 8MB chromedriver.exe. It seems like a bunch of tests of the Model classes made it into a file for Service tests.  I'm not sure how to best test the delete methods (can we see whether the delete method on the repo/service was called?), but the current tests of delete methods don't seem to do anything except make sure no exceptions are thrown.

**Comment by KRT0335** [comment]
I believe I covered the required changes. Respond if there are more mishaps or something that needs to be covered.

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java (hunk 1 above)
Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: Added some assertions to cover this by asserting based on getters and similar methods.

File: src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java (hunk 2 above)
Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java]: Re-added the test but ignored it due to the test not being able with addImage's exception handling.

File: src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java (hunk 3 above)
Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationServiceTests.java]: Was removed, originally made to showcase testing in the question package.
Ultimately showing how flawed the testing for exceptions occurring within another method is.

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java [hunk 4]
```diff
@@ -222,23 +312,28 @@ public void testUpdateTag() {
...
+    Tag exampleTag = new Tag();
+    exampleTag.setId(1);
+    exampleTag.setName("JavaScript");
+
+    when((tagServiceMock.create(exampleTag))).thenReturn(exampleTag);
+    tagServiceMock.create(exampleTag);
+    doNothing().when(tagServiceMock).delete(exampleTag.getId());
+    tagServiceMock.delete(exampleTag.getId());
+    when((tagServiceMock.getById(exampleTag.getId()))).thenReturn(null);
+    tagControllerImpl.deleteTag(exampleTag.getId());
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java:336]: @KRT0335 , this is an example of what I'm referencing. There are no assertions, and it's not evident what you're testing.

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java (hunk 1 above)
Comment (russchristina) [reply src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: @KRT0335 , I still don't see any assertions.

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java (hunk 4 above)
Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java:336]: `Assert.assertNull(tagServiceMock.getById(exampleTag.getId()));`
Was added in the following line.

Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java:336]: Mostly in the mindset that the tag no longer exists after deleting it.
It should be in the code after commit 3d3d7e18a9ec3a793e4fb7599b1876ba14958969
"made changes to tests, ignore one exception that can be tested, added…"

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java [hunk 5]
```diff
@@ -337,51 +613,55 @@ public void testUpdateQuestion() throws QuestionConflictException, QuestionNotFo
...
+
+    // actual update
     System.out.println(exampleUser);
     exampleUser.setUsername("blahblahblah");
     exampleUser.setPassword("djfjgjofoo");
     exampleUser.setExpert(true);
-    
+
     when((userServiceMock.update(exampleUser))).thenReturn(exampleUser);
     System.out.println(exampleUser);
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: @KRT0335 , please remove the sysout...

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java [hunk 6]
```diff
@@ -390,14 +670,13 @@ public void testAddingUser() {
     exampleUser.setId(index);
     exampleUser.setUsername("blah");
     exampleUser.setPassword("dsafjawjf");
-    
+
     when((userServiceMock.create(exampleUser))).thenReturn(exampleUser);
     System.out.println(exampleUser);
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: @KRT0335 , here's another one.

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java [hunk 7]
```diff
@@ -452,7 +731,6 @@ public void testCreatingOrUpdatingUser_Update() {
     assertEquals(anotherExampleUser, userServiceMock.createOrUpdate(anotherExampleUser));
     System.out.println(anotherExampleUser);
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: @KRT0335 , and yet another one.

File: src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java [hunk 8]
```diff
@@ -0,0 +1,614 @@
...
+
+  static Question testQuestion1;
+
+  static Question testQuestion1PostCreate;
+
+  static Question testQuestion2;
+
+  static Page<Question> returnQuestionsPage;
+
+  static {
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java:65]: @KRT0335 , what is the purpose of using these static references and the static initializer? Where in this suite do you use the objects they reference? You just appear to create new mock data in each individual test.

Comment (russchristina) [reply src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java:65]: @KRT0335 , alright. I now see how you've use them, but this abstraction looks a little crazy, no?

File: src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java (hunk 5 above)
Comment (KRT0335) [reply src/test/java/com/revaturelabs/ask/AskApplicationControllerTests.java]: Those were not mine lol

File: src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java [hunk 9]
```diff
@@ -0,0 +1,614 @@
...
+    assertFalse(testUserNullTags.equals(testUserNullFields) & testUserNullTags.equals(testUser));
+    assertFalse(
+        testUserDiffQuestions.equals(testUserNullFields) & testUserDiffQuestions.equals(testUser));
+    assertFalse(
+        testUserNullQuestions.equals(testUserNullFields) & testUserNullQuestions.equals(testUser));
+    assertFalse(
+        testUserDiffResponses.equals(testUserNullFields) & testUserDiffResponses.equals(testUser));
+    assertFalse(
+        testUserNullResponses.equals(testUserNullFields) & testUserNullResponses.equals(testUser));
+  }
```
Comment (russchristina) [inline src/test/java/com/revaturelabs/ask/AskApplicationModelTests.java:613]: This test suite with the liberal use of static references was written during a previous sprint. It needs to be refactored.

**Review by russchristina — APPROVED** [review]

**Pull request merged**